packages/grid_client/tests/modules/applications/casperlabs.test.ts

```diff
@@ -0,0 +1,267 @@
+import axios from "axios";
+import { setTimeout } from "timers/promises";
+
+import { FilterOptions, GatewayNameModel, generateString, GridClient, MachinesModel, randomChoice } from "../../../src";
+import { config, getClient } from "../../client_loader";
+import { bytesToGB, generateInt, getOnlineNode, log, splitIP } from "../../utils";
+
+jest.setTimeout(1500000);
+
+let gridClient: GridClient;
+let deploymentName: string;
+
+beforeAll(async () => {
+  gridClient = await getClient();
+  deploymentName = generateString(15);
+  gridClient.clientOptions.projectName = `casperlabs/${deploymentName}`;
+  gridClient._connect();
+  return gridClient;
+});
+
+//Private IP Regex
+const ipRegex = /(^127\.)|(^10\.)|(^172\.1[6-9]\.)|(^172\.2[0-9]\.)|(^172\.3[0-1]\.)|(^192\.168\.)/;
+
+test("TC2683 - Applications: Deploy Casperlabs", async () => {
+  /**********************************************
+     Test Suite: Grid3_Client_TS (Automated)
+     Test Cases: TC2683 - Applications: Deploy Casperlabs
+     Scenario:
+        - Generate Test Data/casperlabs Config/Gateway Config.
+        - Select a Node To Deploy the casperlabs on.
+        - Select a Gateway Node To Deploy the gateway on.
+        - Deploy the casperlabs solution.
+        - Assert that the generated data matches
+          the deployment details.
+        - Pass the IP of the Created casperlabs to the Gateway
+          Config.
+        - Deploy the Gateway.
+        - Assert that the generated data matches
+          the deployment details.
+        - Assert that the Gateway points at the IP
+          of the created casperlabs.
+        - Assert that the returned domain is working
+          and returns correct data.
+    **********************************************/
+
+  //Test Data
+  const name = "gw" + generateString(10).toLowerCase();
+  const tlsPassthrough = false;
+  let cpu = generateInt(1, 4);
+  let memory = generateInt(1, 6);
+  let rootfsSize = generateInt(0, 2);
```

Comment:
Minimum resources should be as per the solution requirements. Please remove the generateInt and use the resources specified in the instance capacity.

Reply:
root fs should be dynamic depending on the CPU & memory.